modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/RoleResourceTest.java

```diff
@@ -56,6 +57,16 @@ public void setUp() throws Exception {
 		_user = UserTestUtil.addGroupAdminUser(testGroup);
 	}
 
+	@Override
+	@Test
+	public Role testGetRoleByExternalReferenceCode_addRole() throws Exception {
+		Role role = _addRole(RoleConstants.TYPE_REGULAR);
+
+		role.setExternalReferenceCode(role.getName());
```

Comment:
I think this is not needed because inside the _addRole method `Role role = randomRole();` and the randomRole contains already the `role.setExternalReferenceCode(role.getName());`

Remember that if this can me removed you can also simplify the method avoid storing the _addRole in a variable

Reply:
That's true, but it does not explicitly return that random Role, after that there is a line _addRole(role) which further manipulates the role given back by randomRole(). As a consequence the ERC has to be set again, otherwise it will be null. I have tested this. So it seems that that method _addRole() does not exactly works like one would expect.